### ISSUE TITLE
Not set proprierty when using withQueryParams in functional api

### ISSUE THREAD

**Issue by andrefelipe18** [livewire/volt#119]
### Volt Version

1.6.5

### Laravel Version

11.27.2

### PHP Version

8.3

### Database Driver & Version

sqlite

### Description

Hi!

I'm finding one error when i testing my volt component with URL property, (Volt functional API / Pest)

I'm finding one error when i testing my volt component with URL property, when a pass withQueryParams and make dd() in component the property not change the value to value passed in withQueryParams.

My property

![Image](https://github.com/user-attachments/assets/1d9491e6-b594-4e9f-b8d7-41492cc32c9a)

The test with debug

![Image](https://github.com/user-attachments/assets/161ceb9b-ac28-488a-b73a-26c1dbd07566)

Result:

![Image](https://github.com/user-attachments/assets/eb149479-d35d-4265-b126-28a7d4e0eb61)

### Steps To Reproduce

Create a url property and test it withQueryParams()

![Image](https://github.com/user-attachments/assets/7627c143-6708-4669-a3aa-afde08321431)

![Image](https://github.com/user-attachments/assets/8181676f-9aaa-418e-b93e-f0d0836388e0)

**Comment by gcpartyanimal** [comment]
Root-caused this: it is a boot-order bug in `VoltManager`, still present on v1.10.5 and current `main` (the relevant source is identical). Reproduced on Volt v1.10.5 / Livewire v4.3.3 / Laravel v12.63.0 / PHP 8.4.23.

**TL;DR:** `Volt::withQueryParams()` writes the params to a *different* `LivewireManager` instance than the one `Volt::test()` reads from, so they are silently dropped.

### Mechanism

1. `VoltManager` is bound as a singleton in `VoltServiceProvider::register()` ([VoltServiceProvider.php#L27](https://github.com/livewire/volt/blob/96977f2dbcfc6ca736f464006e8ef0064d1493f6/src/VoltServiceProvider.php#L27)) and constructor-injects `Livewire\Volt\LivewireManager` ([VoltManager.php#L23-L27](https://github.com/livewire/volt/blob/96977f2dbcfc6ca736f464006e8ef0064d1493f6/src/VoltManager.php#L23-L27)).
2. But that manager class is only bound as a singleton (and aliased to `'livewire'`) inside an `$app->booted()` callback — [VoltServiceProvider.php#L51-L55](https://github.com/livewire/volt/blob/96977f2dbcfc6ca736f464006e8ef0064d1493f6/src/VoltServiceProvider.php#L51-L55) → [`bindLivewireManager()`](https://github.com/livewire/volt/blob/96977f2dbcfc6ca736f464006e8ef0064d1493f6/src/VoltServiceProvider.php#L61-L73).
3. Every app installed via `volt:install` gets an `App\Providers\VoltServiceProvider` whose `boot()` calls `Volt::mount([...])` ([VoltServiceProvider.stub#L21-L27](https://github.com/livewire/volt/blob/96977f2dbcfc6ca736f464006e8ef0064d1493f6/stubs/VoltServiceProvider.stub#L21-L27)). Provider `boot()` methods run **before** `$app->booted()` callbacks fire, so this first touch of the facade constructs the `VoltManager` singleton pre-boot — and since `Livewire\Volt\LivewireManager` has no container binding yet at that moment, the container auto-resolves a fresh, unmanaged instance into the constructor.
4. After boot, `bindLivewireManager()` binds a **different** instance as the `'livewire'` singleton.
5. Net effect: `Volt::withQueryParams($params)` stores `queryParamsForTesting` on the orphaned pre-boot instance ([VoltManager.php#L81-L86](https://github.com/livewire/volt/blob/96977f2dbcfc6ca736f464006e8ef0064d1493f6/src/VoltManager.php#L81-L86)), while `Volt::test($name)` → `Livewire::test($name)` ([VoltManager.php#L75](https://github.com/livewire/volt/blob/96977f2dbcfc6ca736f464006e8ef0064d1493f6/src/VoltManager.php#L75)) reads through the facade from the post-boot singleton, whose `queryParamsForTesting` is empty. The component's `->url()` / `#[Url]` property keeps its default.

(`Volt::actingAs()` proxies through the same orphan but happens to work, because the base manager's `actingAs()` delegates to the **static** `Testable::actingAs()` rather than storing instance state.)

### Why the package's own test doesn't catch this

[`it('can have query parameters')`](https://github.com/livewire/volt/blob/96977f2dbcfc6ca736f464006e8ef0064d1493f6/tests/Feature/FunctionalComponentTest.php#L127-L139) passes because the suite calls `Volt::mount()` in a `beforeEach` *inside the test file* ([FunctionalComponentTest.php#L23-L28](https://github.com/livewire/volt/blob/96977f2dbcfc6ca736f464006e8ef0064d1493f6/tests/Feature/FunctionalComponentTest.php#L23-L28)) — i.e. after testbench has fully booted the app. `VoltManager` is then constructed post-boot, so its injected manager **is** the bound singleton and writer/reader agree. Only apps that mount from a provider's `boot()` — exactly what `volt:install` publishes — hit the orphan path. A regression test therefore needs to call `Volt::mount()` from a (workbench) provider's `boot()` rather than from inside the test file.

### Repro

Fresh app (`laravel new`, `composer require livewire/volt`, `php artisan volt:install`) — verified in an app whose `App\Providers\VoltServiceProvider` is byte-identical to the published stub.

`resources/views/livewire/volt-qp-repro.blade.php`:

```blade
<?php

use function Livewire\Volt\state;

state(['search' => ''])->url();

?>

<div>
    <p>search: "{{ $search }}"</p>
</div>
```

`tests/Feature/VoltWithQueryParamsReproTest.php`:

```php
<?php

use Livewire\Livewire;
use Livewire\Volt\Volt;

test('bug: Volt::withQueryParams params are silently dropped', function () {
    Volt::withQueryParams(['search' => 'apple'])
        ->test('volt-qp-repro')
        ->assertSet('search', 'apple');
});

test('control: Livewire::withQueryParams hydrates the same component', function () {
    Livewire::withQueryParams(['search' => 'apple']);

    Volt::test('volt-qp-repro')
        ->assertSet('search', 'apple');
});
```

Output:

```
FAIL  Tests\Feature\VoltWithQueryParamsReproTest
⨯ bug: Volt::withQueryParams params are silently dropped
✓ control: Livewire::withQueryParams hydrates the same component

FAILED  Tests\Feature\VoltWithQueryParamsReproTest > bug: Volt::withQueryParams params are silently dropped
Failed asserting that two strings are equal.
-'apple'
+''
```

### Workaround

Set the params on the base Livewire manager instead — this reaches the post-boot singleton that `Volt::test()` actually reads:

```php
Livewire::withQueryParams(['search' => 'apple']);

Volt::test('volt-qp-repro')->assertSet('search', 'apple'); // passes
```

### Suggested fix

Read the live manager the same way `test()` already does:

```diff
     public function withQueryParams(array $params): static
     {
-        $this->manager->withQueryParams($params);
+        Livewire::withQueryParams($params);

         return $this;
     }
```
